src/core/ayayaa-server/ayayaa-logging/Interfaces/IWriter.cs

```diff
@@ -0,0 +1,30 @@
+﻿using ayayaa.logging.Enums;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ayayaa.logging.Interfaces
+{
+    public interface IWriter
+    {
+        /// <summary>
+        /// Takes the given text and priority, formats it into a proper message for the given writer and finally logs it.
+        /// </summary>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        void WriteMessage(string text);
+
+        /// <summary>
+        /// Internal use only. This function converts the message into a value that's usable to the writer.
+        /// (e.g. byte[] for sending it to a remote host.)
+        /// </summary>
+        /// <param name="message"></param>
+        object SerializeMessage(string message);
```

Comment:
"Internal use only" is exactly the opposite of having an interface. I think it's better to just remove those functions from the interface and keep them as implementation detail.